source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -67,9 +67,8 @@ void SslSocket::setTransportSocketCallbacks(Network::TransportSocketCallbacks& c
     provider->registerPrivateKeyMethod(rawSsl(), *this, callbacks_->connection().dispatcher());
   }
 
-  // TODO(fcoras): consider using BIO_s_mem or a BIO with custom read/write functions instead of a
-  // socket BIO which relies on access to a file descriptor.
-  BIO* bio = BIO_new_socket(callbacks_->ioHandle().fdDoNotUse(), 0);
+  // Use custom BIO that reads from/writes to IoHandle
+  BIO* bio = BIO_new_io_handle(&callbacks_->ioHandle());
```

Comment:
Okay, let me add that. But, since I'm not familiar with them, will switching be possible at runtime or the code will have to be recompiled with the feature moved from enabled to disabled list?

Reply:
My understanding is that it is controllable via reloadable config.  The default is to use the new code path.  Since this call only happens once per connection, during initialization of the transport socket object, it is safe for the runtime feature to change during runtime.